packages/console/src/containers/AppContent/components/Topbar/TenantSelector/index.tsx

```diff
@@ -1,41 +1,47 @@
-import { type TenantInfo, TenantTag } from '@logto/schemas/models';
+import { type TenantInfo } from '@logto/schemas/models';
 import classNames from 'classnames';
-import { useRef, useState } from 'react';
+import { useContext, useEffect, useMemo, useRef, useState } from 'react';
 import { toast } from 'react-hot-toast';
 import { useTranslation } from 'react-i18next';
 
 import KeyboardArrowDown from '@/assets/icons/keyboard-arrow-down.svg';
 import PlusSign from '@/assets/icons/plus.svg';
 import Tick from '@/assets/icons/tick.svg';
+import { useCloudSwr } from '@/cloud/hooks/use-cloud-swr';
 import CreateTenantModal from '@/cloud/pages/Main/TenantLandingPage/TenantLandingPageContent/CreateTenantModal';
 import AppError from '@/components/AppError';
+import { TenantsContext } from '@/contexts/TenantsProvider';
 import Divider from '@/ds-components/Divider';
 import Dropdown, { DropdownItem } from '@/ds-components/Dropdown';
-import useTenants from '@/hooks/use-tenants';
 import { onKeyDownHandler } from '@/utils/a11y';
 
 import TenantEnvTag from './TenantEnvTag';
 import * as styles from './index.module.scss';
 
 function TenantSelector() {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  const {
-    tenants,
-    currentTenant: currentTenantInfo,
-    currentTenantId,
-    error,
-    mutate,
-  } = useTenants();
+  const { data: availableTenants, error, mutate } = useCloudSwr('/api/tenants');
+  const { tenants, setTenants, currentTenantId } = useContext(TenantsContext);
 
   const anchorRef = useRef<HTMLDivElement>(null);
   const [showDropdown, setShowDropdown] = useState(false);
   const [showCreateTenantModal, setShowCreateTenantModal] = useState(false);
 
+  useEffect(() => {
+    if (availableTenants) {
+      setTenants(availableTenants);
+    }
+  }, [availableTenants, setTenants]);
```

Comment:
Also here, I think we can remove this when we have `useCloudSwr`. No need to keep tenants in context and maintain the context any more